pageserver/src/layered_repository.rs

```diff
@@ -1611,8 +1611,15 @@ impl LayeredTimeline {
 
         let global_layer_map = GLOBAL_LAYER_MAP.read().unwrap();
         if let Some(oldest_layer) = global_layer_map.get(&layer_id) {
+            let last_lsn = self.get_last_record_lsn();
+            let n_delta_layers = if reconstruct_pages {
+                layers.count_delta_layers(oldest_layer.get_seg_tag(), last_lsn)
+            } else {
+                0
```

Comment:
Yes, no need to count layer if we do no need this information.

Reply:
Let's use an `Option` or `std::lazy::Lazy` instead then. Otherwise it's only a matter of time before someone uses this variable for a different purpose without knowing it could be `0`.

Reply:
Ok, will change to Option